doc/project: document project config arguments and attributes (#1215)
ariga/atlas: doc/md/atlas-schema/projects.md

```hcl
locals {
  tenants  = ["tenant_1", "tenant_2"]
  base_url = "mysql://${var.user}:${var.pass}@${var.addr}"
  
  // Reference local values. 
  db1_url  = "${local.base_url}/db1"
  db2_url  = "${local.base_url}/db2"
}
```

### Data Sources

Data sources enable users to retrieve information stored in an external service or database. The currently supported
data sources are: [`sql`](#data-source-sql).

#### Data source: `sql`

##### Arguments {#data-source-sql-arguments}

- `url` - The [URL](../concepts/url.mdx) of the target database.
- `query` - Query to execute.
- `args` - Optional arguments for any placeholder parameters in the query.

##### Attributes {#data-source-sql-attributes}

- `count` - The number of returned rows.
- `values` - The returned values. e.g. `list(string)`.
- `value` - The first value in the list, or `nil`.

```hcl
data "sql" "tenants" {
  url = var.url
  query = <<EOS
SELECT `schema_name`
  FROM `information_schema`.`schemata`
  WHERE `schema_name` LIKE ?
EOS
  args = [var.pattern]
}

env "prod" {
  // Reference a data source.
  for_each = toset(data.sql.tenants.values)
  url = "${local.base_url}/${each.value}"
}
```

### Environments

> The `env` block defines an environment block that can be selected using the `--env` flag.

Reviewer: ```suggestion
The `env` block defines an environment block that can be selected by using the `--env` flag.
```

##### Arguments {#environment-arguments}

> - `for_each` - A meta-argument that accepts a map or a set of strings and used to compute an `env` instance for each

Reviewer: ```suggestion
- `for_each` - A meta-argument that accepts a map or a set of strings and is used to compute an `env` instance for each
```

set or map item. See the example [below](#multi-environment-example).

- `url` - The [URL](../concepts/url.mdx) of the target database.

- `dev` - The [URL](../concepts/url.mdx) of the [Dev Database](../concepts/dev.md).

- `schemas` - A list of strings donates the schemas that Atlas manages.

- `exclude` - A list of strings defines glob patterns used to filter resources on inspection.

- `migration` - A block defines the migration configuration of the env.
  - `dir` - The [URL](../concepts/url.mdx) to the migration directory.

>   - `revisions_schema` - An optional name to control the schema the revisions table resides in.

Reviewer: ```suggestion
- `revisions_schema` - An optional name to control the schema that the revisions table resides in.
```

- `log` - A block defines the logging configuration of the env per command.
  - `migrate`

>     - `apply` - Set the `--log` flag by setting a custom logging for `migrate apply`.

Reviewer: ```suggestion
    - `apply` - Set the `--log` flag by setting custom logging for `migrate apply`.
```

>     - `lint` - Set the `--log` flag by setting a custom logging for `migrate lint`.

Reviewer: ```suggestion
    - `lint` - Set the `--log` flag by setting custom logging for `migrate lint`.
```

>     - `status` - Set the `--log` flag by setting a custom logging for `migrate status`.

Reviewer: ```suggestion
    - `status` - Set the `--log` flag by setting custom logging for `migrate status`.
```

- `lint` - A block defines the migration linting configuration of the env.

>   - `log` - Override the `--log` flag by setting a custom logging for `migrate lint`.

Reviewer: ```suggestion
  - `log` - Override the `--log` flag by setting custom logging for `migrate lint`.
```

  - `lastest` - A number configures the `--latest` option.

>   - `git.dir` - A path to the repository working directory

Reviewer: ```suggestion
  - `git.dir` - A path to the repository working directory.
```

>   - `git.base` - A run analysis against the base Git branch

Reviewer: ```suggestion
  - `git.dir` - A path to the repository working directory.
```

##### Multi Environment Example

> Atlas adopts the `for_each` meta-argument [Terraform uses](https://www.terraform.io/language/meta-arguments/for_each)

Reviewer: ```suggestion
Atlas adopts the `for_each` meta-argument that [Terraform uses](https://www.terraform.io/language/meta-arguments/for_each)
```

for `env` blocks. Setting the `for_each` argument will compute an `env` block for each item in the provided value. Note
that `for_each` accepts a map or a set of strings.

```hcl
env "prod" {
  for_each = toset(data.sql.tenants.values)
  url      = "${local.base_url}/${each.value}"
  migration {
    dir = "file://migrations"
  }
  log {
    migrate {
      apply = format(
        "{{ json . | json_merge %q }}",
        jsonencode({
          Tenant : each.value
        })
      )
    }
  }
}
```

## Configure Migration Linting

> Project files may declare `lint` blocks to configure how migration linting run in a specific environment or globally.

Reviewer: ```suggestion
Project files may declare `lint` blocks to configure how migration linting runs in a specific environment or globally.
```
